src/api/dependency-discovery/src/dependency-list.js

```diff
@@ -0,0 +1,11 @@
+const { readFile } = require('fs/promises');
+const { join } = require('path');
+const { cwd } = require('process');
+
+async function getDependencies() {
+  const dependencies = await readFile(join(cwd(), 'deps.txt'), 'utf8');
+
+  return [...new Set(dependencies.split('\n').filter((line) => line !== ''))];
```

Comment:
you don't need to use Set here as these are already uniquely sorted by `sort -u`

Reply:
I didn't cover that case. Should I also cover the case when it is `\r` only? I think MacOS uses `\r` but I am not sure. I am going to add it just in case.